### ISSUE TITLE
[Task]: confusing behavior in `AnomalibDataModule._create_test_split()`

### ISSUE THREAD

**Issue by jpcbertoldo** [open-edge-platform/anomalib#980]
### What is the motivation for this task?

The behavior in `AnomalibDataModule._create_test_split()` is confusing, in particular due to this line:

https://github.com/openvinotoolkit/anomalib/blob/4f985851161f2c10eca8ed25f4e69d50571d54c2/src/anomalib/data/base/datamodule.py#L128

Use case: MVTec has a predefined test set that it includes normal images, so I want to use it as it is. I expected the correct config would be `TestSplitMode.NONE` because by "TestSplitMode" I intuitively think that we're talking about a piece of the training set that would be extracted to be used as test. However the correct one is `TestSplitMode.FROM_DIR` because the code does

```
normal_test_data, self.test_data = split_by_label(self.test_data)
# ...
if self.test_split_mode == TestSplitMode.FROM_DIR:
    self.test_data += normal_test_data
```

which (i'm not sure) may change the order of the images in the test set

### Describe the solution you'd like

My proposed behavior:

- `TestSplitMode.NONE` : don't do anything (if normal data exist or not in the dataset, it will remain as it is)
- `TestSplitMode.EXCLUDE_NORMALS`: current behavior of `NONE`
- `TestSplitMode.FROM_DIR`: delete this option
- `TestSplitMode.SYNTHETIC`: as it is now

### Additional context

maybe rename `NONE` to `AS_IS`? or something more intuitive

extra: `TestSplitMode.SYNTHETIC` could have a sub-option to use the training or the test set as a source

**Comment by JeGa** [comment]
I agree, this is very confusing and could lead to wrong test results.
